### ISSUE TITLE
Make Application.getAccountStoreMappings consistent

### ISSUE THREAD

**Issue by dogeared** [stormpath/stormpath-sdk-java#432]
Right now we have:

```
ApplicationAccountStoreMappingList getAccountStoreMappings();
ApplicationAccountStoreMappingList getAccountStoreMappings(Map<String, Object> queryParams);
ApplicationAccountStoreMappingList getApplicationAccountStoreMappings(ApplicationAccountStoreMappingCriteria criteria);
```

Seems like the third one should also be `getAccountStoreMappings`.

We should deprecate `getApplicationAccountStoreMappings` and add `getAccountStoreMappings` that takes an `ApplicationAccountStoreMappingCriteria` as a parameter.

Alternatively, we can make all 3 be `getApplicationAccountStoreMappings`

**Comment by dogeared** [comment]
closes #462

**Comment by jarias** [comment]
Looks like this was already started by @dogeared the work is in 1.0.RC10_in_progress
